src/sentry/static/sentry/app/components/teams/createTeamForm.tsx

```diff
@@ -12,20 +13,18 @@ type Payload = {
 
 type Props = {
   organization: Organization;
-  onSubmit?: (data: object, onSuccess: Function, onError: Function) => void;
+  onSubmit?: (data: Payload, onSuccess: Function, onError: Function) => void;
   onSuccess?: (data: Payload) => void;
   formProps?: Partial<typeof Form>;
 };
 
 export default class CreateTeamForm extends React.Component<Props> {
-  handleCreateTeamSuccess = (data: Payload) => {
-    const {onSuccess} = this.props;
-
-    if (typeof onSuccess !== 'function') {
-      return;
-    }
+  handleSubmit = (data: Record<string, any>, onSuccess: Function, onError: Function) => {
+    callIfFunction(this.props.onSubmit, data as Payload, onSuccess, onError);
+  };
```

Comment:
Wasn't sure of a better way to cast `data` as `Payload`

Reply:
The only thing I can think of would be to use generics for the form to modify `initialData` and outgoing data in the onSbumit? Currently it just has Data as an empty type.

Reply:
Yeah, I think you're right - though not something I'm going to touch in this PR.